### PR TITLE
feat: [log retention improvements pt. 2] add flag for configuring log retention period

### PR DESCRIPTION
**Pull request by tedim52**
## Description
Log Retention Improvements:

This PR allows users to configure log retention period with `--log-retention-period` flag on engine starts/restarts. Default is set to 1 week. 

- [x] PR 1: Introduce `LogFileLayout` and `PerWeekLogFileLayout` and adjust `LogFileManager` and `LogsDatabaseClient` to use this for retrieving log file paths
- [x] PR 2: Make retention configurable via a CLI flag
- [ ] PR 3: Implement `PerHourFileLayout` and swap storage format from `PerWeekFileLayout` to `PerHourFileLayout`
- [ ] PR 4: Make `LogsAggregator` use `LogFileLayout` for determining storage format and set it to use `PerHourFileLayout`
------------------------------

## Is this change user facing?
NO

## References 
https://github.com/kurtosis-tech/kurtosis/issues/2443
https://github.com/kurtosis-tech/kurtosis/issues/2190
https://github.com/kurtosis-tech/kurtosis/pull/2534

### CODE REVIEW

File: cli/cli/commands/engine/restart/restart.go [hunk 1]
```diff
@@ -92,6 +94,13 @@ var RestartCmd = &lowlevel.LowlevelKurtosisCommand{
 			Type:      flags.FlagType_String,
 			Default:   defaultDomain,
 		},
+		{
+			Key:       logRetentionPeriodFlagKey,
```
Comment (h4ck3rk3y) [inline cli/cli/commands/engine/restart/restart.go:98]: is `1w` `1d` 1m` valid; can we add more examples here!

File: cli/cli/commands/engine/restart/restart.go [hunk 2]
```diff
@@ -158,9 +167,18 @@ func run(_ context.Context, flags *flags.ParsedFlags, _ *args.ParsedArgs) error
 		return stacktrace.Propagate(err, "An error occurred while getting the Kurtosis engine enclave manager UI domain name using the flag with key '%v'; this is a bug in Kurtosis", domainFlagKey)
 	}
 
+	logRetentionPeriodStr, err := flags.GetString(logRetentionPeriodFlagKey)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while getting the log retention period string from flag: '%v'", logRetentionPeriodStr)
+	}
+	_, err = time.ParseDuration(logRetentionPeriodStr)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred parsing provided log retention period '%v' into a duration. Ensure the provided value has the proper format. Valid time units are \"ns\", \"us\" (or \"µs\"), \"ms\", \"s\", \"m\", \"h\".", logRetentionPeriodStr)
```
Comment (h4ck3rk3y) [inline cli/cli/commands/engine/restart/restart.go]: oh, not day? what about week?

File: cli/cli/commands/engine/start/start.go [hunk 3]
```diff
@@ -154,16 +163,25 @@ func run(_ context.Context, flags *flags.ParsedFlags, _ *args.ParsedArgs) error
 		return stacktrace.Propagate(err, "An error occurred while getting the Kurtosis engine enclave manager UI domain name using the flag with key '%v'; this is a bug in Kurtosis", domainFlagKey)
 	}
 
+	logRetentionPeriodStr, err := flags.GetString(logRetentionPeriodFlagKey)
```
Comment (h4ck3rk3y) [inline cli/cli/commands/engine/start/start.go:166]: might be over drying; but we can put all of this stuff between commands somewhere

File: docs/docs/cli-reference/engine-restart.md [hunk 4]
```diff
@@ -14,5 +14,7 @@ You may optionally pass in the following flags with this command:
...
 * `--version`: The version (Docker tag) of the Kurtosis engine that should be started. If not set, the engine will start up with the default version.
 * `--enclave-pool-size`: The size of the Kurtosis engine enclave pool. The enclave pool is a component of the Kurtosis engine that allows us to create and maintain 'n' number of idle enclaves for future use. This functionality allows to improve the performance for each new creation enclave request.
+* `--github-auth-token`: The auth token to use for authorizing GitHub operations. If set, this will override the currently logged in GitHub user from `kurtosis github login`, if one exists. Note, this token does not persist when restarting the engine.
+* `--log-retention-period`: The duration in which Kurtosis engine will keep logs for. The engine will remove any logs beyond this period. Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h". (eg. "300ms", "-1.5h" or "2h45m", "168h") The default is set to 1 week (168h). NOTE: Currently, Kurtosis only supports setting retention on weekly intervals. Ongoing work is occurring to make this interval more granular - see https://github.com/kurtosis-tech/kurtosis/pull/2534
```
Comment (h4ck3rk3y) [inline docs/docs/cli-reference/engine-restart.md]: 300 ms is extremely precise :P for logging

**Review by h4ck3rk3y — APPROVED** [review]
LGTM, random ux comments

File: cli/cli/commands/engine/restart/restart.go (hunk 1 above)
Comment (tedim52) [reply cli/cli/commands/engine/restart/restart.go:98]: Makes sense! for context I'm going off whatever `time.ParseDuration` supports https://pkg.go.dev/time#ParseDuration but in the future can add custom logic to support other time periods or use a another library

File: docs/docs/cli-reference/engine-restart.md (hunk 4 above)
Comment (tedim52) [reply docs/docs/cli-reference/engine-restart.md]: oops - ye that was copy pasted from time.ParseDuration documentation but doesn't fit logging context for sure

File: cli/cli/commands/engine/restart/restart.go (hunk 2 above)
Comment (tedim52) [reply cli/cli/commands/engine/restart/restart.go]: as above - right now only using what golangs `time.ParseDuration` supports but can add custom logic later to allow setting other time periods

**Review by barnabasbusa — APPROVED** [review]

**Pull request merged**